packages/decoder/src/decoder.ts

```diff
@@ -0,0 +1,10 @@
+import type { ParseOptions } from "@effect/schema/AST"
+import type * as PR from "@effect/schema/ParseResult"
+
+export interface Decoder<I, O> {
```

Comment:
I've been wondering about this signature a bit..in particular I kept it as a function that is precisely the same signature as the return type of `Parser.validate(schema)`, but I wonder if that's prohibitive. For instance, we can't add anything like `.pipe(...fns)` method onto this function as easily.

❓ What do you think? Do you think the current signature is good enough? Or do you think something like `.pipe` would be a more useful addition when you're composing things together?

Reply:
I think the ability to have a `pipe` method would be nice for concises uses like `D.string.pipe(D.maxLength(5))`

Reply:
Yeah, I was thinking the same. If you don't wanna do it in this PR, I'm cool with that, but if you do, switching to something like makes sense to me. It'll also make it easier to create subtypes like `StructDecoder` and such if we wanna try to improve the union error messages in particular

```ts
interface Decoder<I, O> extends Pipeable {
  readonly decode: (input: I, options?: ParseOptions) => ParseResult<O>
}